src/Symfony/Component/HttpKernel/Tests/HttpCache/ResponseCacheStrategyTest.php

```diff
@@ -138,22 +138,48 @@ public function testLastModifiedIsMergedWithEmbeddedResponse()
     {
         $cacheStrategy = new ResponseCacheStrategy();
 
+        $mainResponse = new Response();
+        $mainResponse->setLastModified(new \DateTimeImmutable('-2 hour'));
+
         $embeddedDate = new \DateTimeImmutable('-1 hour');
+        $embeddedResponse = new Response();
+        $embeddedResponse->setLastModified($embeddedDate);
 
-        // This master response uses the "validation" model
-        $masterResponse = new Response();
-        $masterResponse->setLastModified(new \DateTimeImmutable('-2 hour'));
-        $masterResponse->setEtag('foo');
+        $cacheStrategy->add($embeddedResponse);
+        $cacheStrategy->update($mainResponse);
+
+        $this->assertTrue($mainResponse->headers->has('Last-Modified'));
+        $this->assertSame($embeddedDate->getTimestamp(), $mainResponse->getLastModified()->getTimestamp());
+    }
+
+    public function testLastModifiedIsRemovedWhenEmbeddedResponseHasNoLastModified()
+    {
+        $cacheStrategy = new ResponseCacheStrategy();
+
+        $mainResponse = new Response();
+        $mainResponse->setLastModified(new \DateTimeImmutable('-2 hour'));
 
-        // Embedded response uses "expiry" model
```

Comment:
I think "expiry" refers to the `maxAge`, but we're only using `Last-Modified` in both reqests. `Last-Modified` is, to my understanding, one possibly way of revalidation.